frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,125 @@
+package com.virtuslab.gitmachete.frontend.ui.impl;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.never;
+
+import java.io.PrintWriter;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.SneakyThrows;
+import lombok.val;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GitMacheteErrorReportSubmitter.class})
+public class GitMacheteErrorReportSubmitterTest {
+
+  // Used spy(...) here to still keep the original behaviour of the class and be able to stub getTemplateVariables method
+  // PowerMock API documentation Section 13 [https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/Mockito.html]
+  private static final GitMacheteErrorReportSubmitter gitMacheteErrorReportSubmitter = PowerMockito
+      .spy(new GitMacheteErrorReportSubmitter());
+  private static final LambdaLogger errorReportSubmitterLogger = PowerMockito.mock(LambdaLogger.class);
+  private static String defaultErrorReport;
+  private static String defaultErrorReportWithTemplate;
+  private static String errorReportWithStackTraceAndTitle;
+
+  @BeforeClass
+  @SneakyThrows
+  public static void setUp() {
+    defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
+    defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
+        StandardCharsets.UTF_8).strip();
+    errorReportWithStackTraceAndTitle = IOUtils.resourceToString("/error_report_uri_with_stacktrace_title.txt",
+        StandardCharsets.UTF_8).strip();
+
+    Whitebox.setInternalState(GitMacheteErrorReportSubmitter.class, errorReportSubmitterLogger);
+  }
+
+  private GitMacheteErrorReportSubmitter getMockReportSubmitter() {
+    return getMockReportSubmitter(new HashMap<>());
+  }
+
+  private GitMacheteErrorReportSubmitter getMockReportSubmitter(HashMap<String, String> templateVariables) {
+    PowerMockito
+        .stub(PowerMockito.method(GitMacheteErrorReportSubmitter.class, "getTemplateVariables", IdeaLoggingEvent[].class,
+            String.class))
+        .toReturn(templateVariables);
+
+    return gitMacheteErrorReportSubmitter;
+  }
+
+  private IdeaLoggingEvent[] getMockEvents(String throwableText) {
```

Comment:
Hmmm would it make sense to pass an actual `Throwable` here, so that the stack trace is actually URL-encoded into the resulting URI? the problem might be in the **reproducibility** of this stack trace (if it's always the same regardless of the underlying machine/JDK) tho, up to your judgement

Reply:
I tried doing so, but that's a huge pain in the neck when refactoring as the throwable remembers on which line it was created and so the URI will change everytime somebody edits the code/adds an import. Apart from that, encoding the `Throwable` into the URI is just String operations so I wouldn't expect any problems with that other than #1008